profiling/src/profile/mod.rs

```diff
@@ -659,47 +666,77 @@ impl Profile {
             local_root_span_id_label_offset,
         };
 
-        let id = match self.samples.get_index_of(&s) {
-            None => {
-                self.samples.insert(s, values);
-                SampleId::new(self.samples.len() - 1)
+        let id = if let Some(ts) = timestamp {
+            match self.timestamped_samples.get_full_mut(&s) {
+                None => {
+                    let series = vec![(ts, values)];
+                    let (idx, _) = self.timestamped_samples.insert_full(s, series);
+                    SampleId::timestamped(idx)
+                }
+                Some((idx, _, series)) => {
+                    let last_ts = series.last().unwrap().0;
+                    anyhow::ensure!(
+                        ts > last_ts,
+                        "Timestamps should be monotonically increasing, but instead saw {} {}",
+                        last_ts,
+                        ts
+                    );
```

Comment:
How important is this guarantee? I'm not sure it would hold for runtimes like .NET where things are multi-threaded and queued. It seems probable that the order the queued things are processed will not result in monotonic timestamps at some point, even ignoring cases like time going backwards.

Reply:
This guarantee is not necessary to the implementation, (with one caveat).  This was more of a sanity check, that  

The caveat is the question is what to do in the case of duplicated timestamps for the same sample.  Without this check, we would simply record both observations to the sample.  With the check, we would reject it as impossible.  What is the desired behaviour in that case?

Reply:
Aggregating two observations with the same timestamp makes sense to me conceptually -- we're saying that multiple things happened with that same timestamp.

Reply:
I removed the check